pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1881,8 +1881,9 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 		DeprecatedV1:    deprecatedV1,
 	}
 
-	if util.IsIstioVersionGE13(opts.proxy) {
+	if util.IsIstioVersionGE13(opts.proxy) && !opts.isGateway {
```

Comment:
Just check for the proxy type. That’s all you need here.

Reply:
fixed.